src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1040,15 +1038,23 @@ const Json::Value& CascadiaSettings::_GetDisabledProfileSourcesJsonObject(const
 }
 
 // Method Description:
-// - Write the current state of CascadiaSettings to our settings.json file
+// - Write the current state of CascadiaSettings to our settings file
+// - Create a backup file with the current contents
 // Arguments:
 // - <none>
 // Return Value:
 // - <none>
-void CascadiaSettings::Export() const
+void CascadiaSettings::WriteSettingsToDisk() const
 {
+    auto settingsPath{ CascadiaSettings::SettingsPath() };
+
+    // write backup settings file
+    auto backupSettingsPath{ settingsPath + L".backup" };
+    _WriteSettings(_userSettingsString, settingsPath);

```

Comment:
Is there a TODO to remove this?

Reply:
Nope. A part of me wonders if we even want to remove this later? This would be a good conversation to have once Settings UI is on par with JSON though.

Reply:
I really like the idea of backups, though I'd like it even better if it were "previousname-datetime.backup" and just kept rolling. Even more so than taking this feature out. Maybe it's configurable? Keep last 10? But I'm a backup nut.

Reply:
Wait hold up this won't work at all!
1. This needs to be 
```suggestion
    // write backup settings file
    auto backupSettingsPath{ settingsPath + L".backup" };
    _WriteSettings(_userSettingsString, backupSettingsPath);
```
2. If the user saves the settings twice, won't we just blow away their original settings with the one that we re-serialized after the first launch?

Reply:
> 1. This needs to be 

whoops. Fixed (in upcoming commit).

> 2. If the user saves the settings twice, won't we just blow away their original settings with the one that we re-serialized after the first launch?

Yeah, we only hold one backup at a time. We could add multiple backups like Michael was saying ("settings.json-<date><time>.backup"). But won't that take up too much space? (also, how do you get the current date/time?)

Reply:
We could always just not write that file _if it exists_, yea? Presumably the first time, we'll back it up, and then we'll only ever touch the settings-ui modified version

Reply:
@zadjii-msft Done. Not sure if there's a faster way to check if a file exists than `CreateFileW()`. A bit annoyed that we'd call it twice (once to check if the file exists, and again to write it), but I'm not too familiar with this stuff.